static/js/narrow.js

```diff
@@ -248,7 +248,13 @@ exports.activate = function (raw_operators, opts) {
     $('#search_query').val(Filter.unparse(operators));
     search.update_button_visibility();
 
-    compose_actions.on_narrow();
+    // To not open the compose box with using 'p' hotkey to
+    // narrow to next unread PM thread
+    if (raw_operators[0].operator === "pm-with" && opts.hotkey) {
+        compose_actions.on_narrow(true);
```

Comment:
I would pass this parameter around a little bit differently.  We often use the `opts` convention, which makes the calling code more readable.  In this case it's hard to know what the `true` parameter really means, whereas you could say instead say `compose.on_actions({force_close: true})`.  Also, that allows the caller of **this** function to also pass pass `{force_close: true}`, and by letting the caller set the option, we wouldn't need to check the operator here.